modules/dnn/src/layers/dft_layer.cpp

```diff
@@ -0,0 +1,310 @@
+    {
+        dftAlongAxisWorker<T, ComplexVec>(
+            src, dst, dimSizesSrc, stridesSrc, stridesDst,
+            iterDims, outerSizes, outerStep, totalOuter,
+            axis, N, outN, strideAxisSrc, strideAxisDst,
+            matTypeComplex, 0, inverse,
+            [&](Mat& inRow, const T* in, int origLen, int len, size_t stride){
+                ComplexVec* ptr = inRow.ptr<ComplexVec>(0);
+                for (int n = 0; n < origLen; ++n)
+                {
+                    size_t offSrc = (size_t)n * stride;
+                    ptr[n][0] = in[offSrc + 0];
+                    ptr[n][1] = in[offSrc + 1];
+                }
+                const ComplexVec zeroVal(T(0), T(0));
+                for (int n = origLen; n < len; ++n) ptr[n] = zeroVal;
+            }
+        );
+    }
+    else
+    {
+        dftAlongAxisWorker<T, ComplexVec>(
+            src, dst, dimSizesSrc, stridesSrc, stridesDst,
+            iterDims, outerSizes, outerStep, totalOuter,
+            axis, N, outN, strideAxisSrc, strideAxisDst,
+            matTypeReal, DFT_COMPLEX_OUTPUT, inverse,
+            [&](Mat& inRow, const T* in, int origLen, int len, size_t stride){
+                T* ptr = inRow.ptr<T>(0);
+                for (int n = 0; n < origLen; ++n)
+                {
+                    size_t offSrc = (size_t)n * stride;
+                    ptr[n] = in[offSrc];
+                }
+                for (int n = origLen; n < len; ++n) ptr[n] = T(0);
+            }
+        );
+    }
+}
+
+class DFTLayerImpl CV_FINAL : public DFTLayer {
+public:
+    DFTLayerImpl(const LayerParams &params)
+    {
+        setParamsFrom(params);
+        inverse = params.get<int>("inverse", 0) != 0;
+        onesided = params.get<int>("onesided", 0) != 0;
+        axis_attr = params.get<int>("axis", 1);
+        dft_length = params.get<int>("dft_length", -1);
+    }
+
+    virtual bool getMemoryShapes(const std::vector<MatShape> &inputs,
+                                 const int /*requiredOutputs*/,
+                                 std::vector<MatShape> &outputs,
+                                 std::vector<MatShape> &/*internals*/) const CV_OVERRIDE
+    {
+        CV_Assert(inputs.size() >= 1);
+        const MatShape &inshape = inputs[0];
+        MatShape out = inshape;
+
+        if (!out.empty())
+        {
+            int last = out.back();
+            if (last == 1)
+                out.back() = 2;
+            else if (last != 2)
+                out.push_back(2);
+
+            int ndims_in = (int)inshape.size();
+            int ax = axis_attr;
+            if (ax == INT_MIN)
+            {
+                ax = (inshape.back() == 2 || inshape.back() == 1) ? ndims_in - 2 : ndims_in - 1;
+            }
+            if (ax < 0) ax += ndims_in;
+            if (ax >= 0 && ax < (int)out.size() - 1)
+            {
+                int signalLen = dft_length > 0 ? dft_length : (ax < (int)inshape.size() ? inshape[ax] : out[ax]);
```

Comment:
I don't see how dft_length is extracted from the input tensor. It should be the code to extract scalar from the second input if it's constant.